src/main/java/org/opentripplanner/graph_builder/module/osm/naming/SidewalkNamer.java

```diff
@@ -0,0 +1,290 @@
+import org.geotools.geometry.jts.JTS;
+import org.geotools.referencing.CRS;
+import org.geotools.referencing.crs.DefaultGeographicCRS;
+import org.locationtech.jts.geom.Coordinate;
+import org.locationtech.jts.geom.Envelope;
+import org.locationtech.jts.geom.Geometry;
+import org.locationtech.jts.geom.LineString;
+import org.locationtech.jts.geom.MultiLineString;
+import org.locationtech.jts.geom.Point;
+import org.locationtech.jts.operation.buffer.BufferParameters;
+import org.opentripplanner.framework.geometry.GeometryUtils;
+import org.opentripplanner.framework.geometry.HashGridSpatialIndex;
+import org.opentripplanner.framework.geometry.SphericalDistanceLibrary;
+import org.opentripplanner.framework.i18n.I18NString;
+import org.opentripplanner.framework.lang.DoubleUtils;
+import org.opentripplanner.framework.logging.ProgressTracker;
+import org.opentripplanner.graph_builder.module.osm.StreetEdgePair;
+import org.opentripplanner.graph_builder.services.osm.EdgeNamer;
+import org.opentripplanner.openstreetmap.model.OSMWithTags;
+import org.opentripplanner.street.model.edge.StreetEdge;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * A namer that assigns names of nearby streets to sidewalks if they meet certain
+ * geometric similarity criteria.
+ * <p>
+ * The algorithm works as follows:
+ *  - for each sidewalk we look up (named) street edges nearby
+ *  - group those edges into groups where each edge has the same name
+ *  - draw a flat-capped buffer around the sidewalk, like this: https://tinyurl.com/4fpe882h
+ *  - check how much of a named edge group is inside the buffer
+ *  - remove those groups which are below MIN_PERCENT_IN_BUFFER
+ *  - take the group that has the highest percentage (as a proportion of the sidewalk length) inside
+ *    the buffer and apply its name to the sidewalk.
+ * <p>
+ * This works very well for OSM data where the sidewalk runs a parallel to the street and at each
+ * intersection the sidewalk is also split. It doesn't work well for sidewalks that go around
+ * the corner, like https://www.openstreetmap.org/way/1059101564. These cases are, however, detected
+ * by the above algorithm and the sidewalk name remains the same.
+ */
+public class SidewalkNamer implements EdgeNamer {
+
+  private static final Logger LOG = LoggerFactory.getLogger(SidewalkNamer.class);
+  private static final double MIN_PERCENT_IN_BUFFER = .85;
+  private static final int BUFFER_METERS = 25;
+
+  private HashGridSpatialIndex<EdgeOnLevel> streetEdges = new HashGridSpatialIndex<>();
+  private Collection<EdgeOnLevel> unnamedSidewalks = new ArrayList<>();
+  private PreciseBuffer preciseBuffer;
+
+  @Override
+  public I18NString name(OSMWithTags way) {
+    return way.getAssumedName();
+  }
+
+  @Override
+  public void recordEdges(OSMWithTags way, StreetEdgePair pair) {
+    // this way is a sidewalk and hasn't been named yet (and is not explicitly unnamed)
```

Comment:
Capitalize first letter of sentence.

Reply:
https://github.com/opentripplanner/OpenTripPlanner/pull/5774/commits/60c4ee925f51689bacd596f517ea709984b36615